File: Codesignal/Eric/74.TimedReading.js
function solution(ml, text) {
    let arr = text.split(' ');
    let cnt = 0;
    arr.map( x => {
        const match = x.match(/[a-zA-Z]+/g);
        match && (match[0]).length <= ml ? cnt++ : null
    });
    return cnt
}

function solution2(maxLength, text) {
    return (text.match(/\w+/g) || []).filter(elm => elm.length <= maxLength).length
}

console.log(solution(4, "The Fox asked the stork, 'How is the soup?'")); // 7
console.log(solution(1, "...")); // 0
console.log(solution(3, "This play was good for us.")) // 3

File: Codesignal/Eric/74.TimedReading.py
import re

def solution(ml, text):
    result = 0
    arr = text.split(" ")
    for x in arr:
        match = re.findall(r'[a-zA-Z]+', x)
        if match and len(match[0]) <= ml:
            result += 1
    return result


# split 할때 여러개의 delimiter를 동시에 사용할 수 있다.        
def solution2(maxLength, text):
    return sum(1 for i in re.split("[] ;',:?!]",text) if maxLength >= len(i) and i)


def test(maxLength, text):
    return [ i for i in re.split("[] ;',:?!]",text)]

print(solution2(4, "The Fox asked the stork, 'How is the soup?'")) # 7
print(solution2(1, "...")) # 0
print(solution2(3, "This play was good for us.")) # 3